### PR TITLE
SAK-48294 - Commons: Make sure the placeholder text is cleared out before content is "dropped".

### PR DESCRIPTION
**Pull request by maurercw**

### CODE REVIEW

File: commons/tool/src/webapp/js/commons.js [hunk 1]
```diff
@@ -124,6 +124,22 @@ commons.switchState = function (state, arg) {
...
+                if (this.innerHTML == commons.i18n['post_editor_initial_text']) {
+                    this.innerHTML = '';
+                    $('#commons-editor-post-button').prop('disabled', false);
+                    editorPostButton.prop('disabled', false);
+                    editorCancelButton.prop('disabled', false);
+                }
+                editor.focus();
+
+                // get data
+                var dt = e.originalEvent.dataTransfer;
```
Comment (adrianfish) [inline commons/tool/src/webapp/js/commons.js:138]:
```suggestion
                let dt = e.originalEvent.dataTransfer;
```

File: commons/tool/src/webapp/js/commons.js [hunk 2]
```diff
@@ -124,6 +124,22 @@ commons.switchState = function (state, arg) {
...
+                if (!dt) dt = window.dataTransfer;
+                var dropped = dt.getData('text');
```
Comment (adrianfish) [inline commons/tool/src/webapp/js/commons.js:140]:
```suggestion
                const dropped = dt.getData('text');
```

File: commons/tool/src/webapp/js/commons.js [hunk 3]
```diff
@@ -124,6 +124,22 @@ commons.switchState = function (state, arg) {
                     editorCancelButton.prop('disabled', false);
                 }
                 editor.focus();
+            }).on('drop', function (e) {
+                // clear placeholder text
+                if (this.innerHTML == commons.i18n['post_editor_initial_text']) {
+                    this.innerHTML = '';
+                    $('#commons-editor-post-button').prop('disabled', false);
+                    editorPostButton.prop('disabled', false);
```
Comment (adrianfish) [inline commons/tool/src/webapp/js/commons.js:132]: These are the same thing, I think.

**Review by adrianfish — CHANGES_REQUESTED** [review]
Just a few small things.

File: commons/tool/src/webapp/js/commons.js (hunk 3 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js:132]: Are you saying it's the same as the "click" part from line 119 above?  If so, yes, you are correct.  However, the click event doesn't fire when you drag and drop content, so I needed to do the same exact thing for the drop event.

File: commons/tool/src/webapp/js/commons.js (hunk 2 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js:140]: Happy to make this change, as well as the suggestion above, but I copied it from the existing "paste" just below at line 131/147.  I can change those too if desired.

File: commons/tool/src/webapp/js/commons.js (hunk 1 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js]: Change made

File: commons/tool/src/webapp/js/commons.js (hunk 3 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js:132]: Oh, maybe you mean line 131 and 132 are doing the same thing?

File: commons/tool/src/webapp/js/commons.js [hunk 4]
```diff
@@ -119,16 +119,30 @@ commons.switchState = function (state, arg) {
...
+                if (this.innerHTML == commons.i18n['post_editor_initial_text']) {
+                    this.innerHTML = '';
+                    editorPostButton.prop('disabled', false);
+                    editorCancelButton.prop('disabled', false);
+                }
+                editor.focus();
+
+                // get data
+                let dt = e.originalEvent.dataTransfer;
+                if (!dt) dt = window.dataTransfer;
```
Comment (adrianfish) [inline commons/tool/src/webapp/js/commons.js:137]:
```suggestion
                const dt = e.originalEvent.dataTransfer || window.dataTransfer;
```

File: commons/tool/src/webapp/js/commons.js [hunk 5]
```diff
@@ -119,16 +119,30 @@ commons.switchState = function (state, arg) {
...
+                const dropped = dt.getData('text');
+                wrapAndInsert(dropped, true);
+                e.preventDefault();
             }).on('paste', function (e) {
 
-                var cd = e.originalEvent.clipboardData;
+                let cd = e.originalEvent.clipboardData;
                 if (!cd) cd = window.clipboardData;
```
Comment (adrianfish) [inline commons/tool/src/webapp/js/commons.js:144]:
```suggestion
                const cd = e.originalEvent.clipboardData || window.clipboardData;
```

**Review by adrianfish — APPROVED** [review]
Thanks Chris. I've added a couple of commit suggestions just to turn those lets into consts with one line.

File: commons/tool/src/webapp/js/commons.js (hunk 4 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js]: Change made, thanks for the suggestion!

File: commons/tool/src/webapp/js/commons.js (hunk 5 above)
Comment (maurercw) [reply commons/tool/src/webapp/js/commons.js]: Change made.

**Review by adrianfish — APPROVED** [review]
Thanks Chris :)

**Pull request merged**